src/NServiceBus.Metrics/MetricsFeature.cs

```diff
@@ -46,27 +53,62 @@ static void ConfigureMetrics(FeatureConfigurationContext context, MetricsContext
 
     class MetricsReporting : FeatureStartupTask
     {
-        public MetricsReporting(MetricsContext metricsContext, MetricsOptions metricsOptions, IBuilder builder)
+        public MetricsReporting(List<Func<IBuilder, MetricsReport>> reportBulders, MetricsContext metricContext, IBuilder builder, TimeSpan reportingInterval)
         {
-            this.metricsOptions = metricsOptions;
+            this.reportBulders = reportBulders;
+            this.metricContext = metricContext;
             this.builder = builder;
-            metricsConfig = new MetricsConfig(metricsContext);
+            this.reportingInterval = reportingInterval;
+
+            stopping = new CancellationTokenSource();
         }
 
         protected override Task OnStart(IMessageSession session)
         {
-            metricsOptions.SetUpReports(metricsConfig, builder);
+            var reports = reportBulders.Select(rb => rb(builder)).ToArray();
+
+            ReportInIntervals(reports).IgnoreContinuation();
+
             return Task.FromResult(0);
         }
 
+        async Task ReportInIntervals(MetricsReport[] reports)
+        {
+            while (!stopping.Token.IsCancellationRequested)
+            {
+                try
+                {
+                    await Task.Delay(reportingInterval).ConfigureAwait(false);
+
+                    var metricsData = metricContext.DataProvider.CurrentMetricsData;
+                    var dataSnapshot = metricsData.ResetMetrics();
+
+                    foreach (var report in reports)
+                    {
+                        report.RunReport(dataSnapshot, HealthChecks.GetStatus, stopping.Token);
```

Comment:
Should there be a try/catch around this? Otherwise if the first reporter fails, the others won't be attempted.